src/hotspot/cpu/ppc/c1_LIRAssembler_ppc.cpp

```diff
@@ -1070,23 +1070,23 @@ void LIR_Assembler::stack2stack(LIR_Opr src, LIR_Opr dest, BasicType type) {
     case T_FLOAT: {
       Address from = frame_map()->address_for_slot(src->single_stack_ix());
       Address to   = frame_map()->address_for_slot(dest->single_stack_ix());
-      __ lwz(tmp, from.disp(), from.base());
+      __ lwz(tmp, from);
       __ stw(tmp, to.disp(), to.base());
       break;
     }
     case T_ADDRESS:
     case T_OBJECT: {
       Address from = frame_map()->address_for_slot(src->single_stack_ix());
       Address to   = frame_map()->address_for_slot(dest->single_stack_ix());
-      __ ld(tmp, from.disp(), from.base());
+      __ ld(tmp, from);
       __ std(tmp, to.disp(), to.base());
       break;
     }
     case T_LONG:
     case T_DOUBLE: {
       Address from = frame_map()->address_for_double_slot(src->double_stack_ix());
       Address to   = frame_map()->address_for_double_slot(dest->double_stack_ix());
-      __ ld(tmp, from.disp(), from.base());
+      __ ld(tmp, from);
       __ std(tmp, to.disp(), to.base());
```

Comment:
why these store instruction left behind ? 

Can't we just pass address and give it scratch register to play with ?

Reply:
Oh, we don't even need `tmp` and it defaults to `noreg`. Even better.

Reply:
Thanks for suggestion! I have adapted the changes for store instructions. Testing (tier1) on linux-ppc64le is successfully completed. Testing to be done for aix-ppc.